### DISCUSSION TITLE
Any idea on using wire:loading on dispatch?

### DISCUSSION THREAD

**Discussion by whousehomestay** [livewire/livewire#7287] [1. Help] [answered]
When we dispatch an event on js using Livewire.dispatch, is there any idea to use wire:loading and show loading indicator when this dispatch is processed?

**Comment by PhiloNL** [comment, marked as answer]
Events are processed async and there is no way of telling what events are doing (e.g. make a network request or simply run some javascript). You could use something like Laravel Echo and have a loading indicator show and hide based on back-end events, but you will have to build this logic yourself or find a way to hook into wire:loading.

**Reply by whousehomestay** [reply]
Oww... Then I made a misunderstanding here. So, Livewire.dispatch('action-name', {}) isn't always perform a network request?
Ok. Get it. Thank you for the answer.

**Comment by akulmehta** [comment]
Not sure if this helps but here were my workarounds. 

You can add the wire:loading to the component which is receiving the dispatched event.

Alternatively, you can create a separate component whose sole purpose is to show the "Loading..." message and have the listener for this as the dispatched event.
